.github/workflows/reusable_testing.yml

```diff
@@ -0,0 +1,48 @@
+name: reusable_testing
+
+on:
+  workflow_call:
+    inputs:
+      artifact_name:
+        description: 'The name of the tarantool build artifact'
+        default: ubuntu-focal
+        required: false
+        type: string
+
+jobs:
+  run_tests:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: 'Clone the smtp module'
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.repository_owner }}/smtp
+
+      - name: 'Download the tarantool build artifact'
+        uses: actions/download-artifact@v2
+        with:
+          name: ${{ inputs.artifact_name }}
+      # TODO(ylobankov): Just in case remove tarantool that could be previously
+      # installed on the runner. Relevant to self-hosted runners. Remove this
+      # step when only GitHub-hosted runners are used for integration testing.
+      - run: sudo apt-get -y purge 'tarantool*'
+      - name: 'Install tarantool'
+        # TODO(ylobankov): Install package dependencies. Now we're lucky: all
+        # dependencies are already there.
+        run: sudo dpkg -i tarantool*.deb
+
+      - name: 'Install smtp'
+        run: tarantoolctl rocks make
+
+      - run: cmake .
+      - run: make check
```

Comment:
Either build it using `tarantoolctl rocks make` (and run tests using `make -C build.luarocks check`) or build it using `cmake . && make` (and run tests using just `make check`). If you doing both actions, you just doing it twice.

Any of the ways is okay for me, but I would suggest to stick with the simpler one.

Reply:
Done.